### PR TITLE
test[CL]: createPosition table driven tests (4/6)

### PR DESCRIPTION
**Pull request by czarcas7ic**
Partially Closes: #3157 

## What is the purpose of the change

Adds table driven test for createPosition

Fulfills following requirements as laid out in the issue:
- [x] tests added
- [x] coverage maximized
- [x] godocs added
- [x] aim for 1 method per PR

### CODE REVIEW

File: x/concentrated-liquidity/lp.go [hunk 1]
```diff
@@ -16,8 +16,11 @@ import (
 // to create position if the desired amounts cannot be satisfied.
 // On success, returns an actual amount of each token used and liquidity created.
 // Returns error if:
-// TODO: list error cases
-// TODO: table-driven tests
+// - the provided ticks are out of range / invalid
+// - the pool provided does not exist
+// - the liquidity delta is zero
+// - the amount0 or amount1 returned from the position update is less than the defined minimums
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp.go:22]: I got confused thinking that we defined them in code but these are given as input

```suggestion
// - the amount0 or amount1 returned from the position update is less than the given minimums
```

File: x/concentrated-liquidity/lp.go [hunk 2]
```diff
@@ -34,6 +37,7 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
+	// TODO: I was unable to create a test case that outputs a liquidity delta of zero, so this error path is untested
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp.go:40]: What if `amount0Desired` and `amount1Desired` are both 0?

File: x/concentrated-liquidity/lp_test.go [hunk 3]
```diff
@@ -24,42 +24,52 @@ type lpTest struct {
...
 			amount1Minimum: baseCase.amount1Expected.Mul(sdk.NewInt(2)),
 			expectedError:  types.InsufficientLiquidityCreatedError{Actual: baseCase.amount1Expected, Minimum: baseCase.amount1Expected.Mul(sdk.NewInt(2))},
 		},
-		"error: invalid tick": {
-			lowerTick:     types.MaxTick + 1,
-			expectedError: types.InvalidTickError{Tick: types.MaxTick + 1, IsLower: true},
-		},
 		// TODO: add more tests
 		// - custom hand-picked values
-		// - error edge cases
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:62]: Can we make an issue to track the remaining TODOs please? I don't think these are tracked anywhere

File: x/concentrated-liquidity/lp_test.go [hunk 4]
```diff
@@ -69,43 +79,65 @@ func (s *KeeperTestSuite) TestCreatePosition() {
...
+			// - some error was emitted
+			// - asset0 and asset1 that were calculated from create position is nil
+			// - the error emitted was the expected error
+			// - account balances are untouched
 			if tc.expectedError != nil {
 				s.Require().Error(err)
 				s.Require().Equal(asset0, sdk.Int{})
 				s.Require().Equal(asset1, sdk.Int{})
-				s.Require().ErrorAs(err, &tc.expectedError)
+				s.Require().ErrorContains(err, tc.expectedError.Error())
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:113]: q: Why did `ErrorAs` not work anymore?

File: x/concentrated-liquidity/lp_test.go [hunk 5]
```diff
@@ -24,42 +24,52 @@ type lpTest struct {
...
+		liquidityAmount: DefaultLiquidityAmt,
 	}
 )
 
 func (s *KeeperTestSuite) TestCreatePosition() {
 	tests := map[string]lpTest{
-		// "base case": *baseCase,
-		"amount of token 0 is smaller than minimum; should fail and not update state": {
+		"base case": {},
+		"error: non-existent pool": {
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:46]: Do you mind checking if it makes sense to port these edge case tests for `withdrawPosition` as well? They should be very similar

File: x/concentrated-liquidity/lp_test.go (hunk 4 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/lp_test.go:113]: Oh it does, just changed it back (was having issues not realizing it was supposed to be a pointer)

File: x/concentrated-liquidity/types/errors.go [hunk 6]
```diff
@@ -154,3 +154,10 @@ type InvalidPriceLimitError struct {
 func (e InvalidPriceLimitError) Error() string {
 	return fmt.Sprintf("invalid sqrt price limit given (%s), should be greater than (%s) and less than (%s)", e.SqrtPriceLimit, e.LowerBound, e.UpperBound)
 }
+
+type ZeroLiquidityError struct {
+}
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/errors.go:159]: If there is nothing to format, we could define this as a regular error with `errors.New(...)` right?

**Review by p0mvn — APPROVED** [review]
🚢
